server-core/engine/src/main/java/io/infectnet/server/engine/core/world/strategy/AStarPathFinderStrategy.java

```diff
@@ -0,0 +1,298 @@
+
+    /**
+     * The destination Position.
+     */
+    private Position targetPos;
+
+    /**
+     * The list of Nodes that we do not yet consider fully searched.
+     */
+    private List<Node> openNodes;
+
+    /**
+     * Creates a startegy with the given heuristic to use in the A* algorithm.
+     * @param heuristic the specific heuristic function to use
+     */
+    public AStarPathFinderStrategy(Heuristic heuristic){
+        this.heuristic = heuristic;
+    }
+
+    @Override
+    public LinkedList<Tile> findPath(World world, Position start, Position target) {
+
+        /* Initialization */
+
+        this.world = world;
+        targetPos = target;
+        openNodes = new ArrayList<>();
+
+        /* The list of nodes that have been searched through. */
+
+        List<Node> closedNodes = new ArrayList<>();
+
+        /* Finding the right target, if the given is a occupied by an Entity,
+        which is almost always e.g. resource, other player's entity. */
+
+        targetPos = resetTarget();
+
+        /* Putting the starting node with its parameters in the list of open Nodes,
+         to expand it later on. */
+
+        Node startNode = new Node(start);
+        startNode.setCost(0);
+        startNode.setHeuristic(heuristic.heuristic(world, start, targetPos));
+        startNode.setSumCost(startNode.getHeuristic());
+
+        openNodes.add(startNode);
+
+        /* The current Node represents the current end of the found path so far. */
+
+        Node current = null;
+
+        /* The loop goes as long as there are nodes to expand
+        or we were able to find a path to the target. */
+
+        while(!openNodes.isEmpty()){
+
+            /* The current node should be chosen from the openNodes list. */
+
+            current = findNodeToExpand();
+
+            /* If we got to the desired target, the algorithm stops. */
+
+            if(current.getPosition().equals(targetPos)){
+                break;
+            }
+
+            /* Before we expand the cosen Node named as current,
+            we switch its formed place in the open list to a place in the closed nodes list. */
+
+            openNodes.remove(current);
+            closedNodes.add(current);
+
+            /* Expanding the current node */
+
+            for (Node neighbour : current.getNeighbours()) {
+                if (neighbour == null) {
+                    continue;
+                }
+
+                /* Computing the cost of the current Node, and comparing with its neighbour's cost,
+                 to decide where to continue the path building.*/
+
+                int nextCost = current.getCost() + calculateCostOfTile(neighbour);
+
+                if (nextCost < neighbour.getCost()) {
+                    openNodes.remove(neighbour);
+                    closedNodes.remove(neighbour);
+                }
+
+                /* The only case where all conditions are met */
+
+                if (!openNodes.contains(neighbour)
```

Comment:
The condition in this `if` is pretty hard to read. Can you extract it into a separate method with a descriptive name? That'd greatly increase the readability of this section.